chain/consensus/hierarchical/subnet/submgr/manager.go

```diff
@@ -455,6 +462,8 @@ func (s *Service) AddSubnet(ctx context.Context, params *hierarchical.SubnetPara
 		return address.Undef, aerr
 	}
 
+	stats.Record(ctx, metrics.SubnetCreatedCount.M(1))
```

Comment:
I don't think this is is right. Instead of counting every time we create a new subnet, I would keep polling the state periodically (as you do below) and update the metrics "real-time" accordingly. Directly from the state we can get the number of subnets created, active an inactive, no need to count here.

Reply:
Unless we want to track the total amount of subnets created in the history of the subnet, in which case this is correct :)